scripts/solhint-custom/index.js

```diff
@@ -71,14 +71,18 @@ module.exports = [
     }
   },
 
-  // TODO: re-enable and fix
-  // class extends Base {
-  //   static ruleId = 'no-external-virtual';
-  //
-  //   FunctionDefinition(node) {
-  //     if (node.visibility == 'external' && node.isVirtual) {
-  //       this.error(node, 'Functions should not be external and virtual');
-  //     }
-  //   }
-  // },
+  class extends Base {
+    static ruleId = 'no-external-virtual';
+
+    FunctionDefinition(node) {
+      // Skip constructor functions and interface definitions
+      if (node.isConstructor || node.parent.kind === 'interface') {
+        return;
+      }
+      
+      if (node.visibility === 'external' && node.isVirtual) {
```

Comment:
This causes a conflict with prettier that requires the `receive` function to be external.